src/main/java/application/onlinebookstore/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,111 @@
+        if (scByUserId.isEmpty()) {
+            ShoppingCart shoppingCart = new ShoppingCart();
+            shoppingCart.setUser(getUserById(userId));
+            shoppingCart.setCartItems(new HashSet<>());
+            ShoppingCart savedShoppingCart = shoppingCartRepository.save(shoppingCart);
+            Long id = savedShoppingCart.getId();
+            CartItemDto savedCartItem = cartItemService.save(cartItemDtoRequest, id);
+            savedShoppingCart.getCartItems().add(cartItemMapper.toModel(savedCartItem));
+            return shoppingCartMapper.toDto(shoppingCartRepository.save(savedShoppingCart));
+        }
+        ShoppingCart shoppingCart = scByUserId.get();
+        CartItemDto savedCartItem = cartItemService.save(cartItemDtoRequest, shoppingCart.getId());
+        shoppingCart.getCartItems().add(cartItemMapper.toModel(savedCartItem));
+        return shoppingCartMapper.toDto(shoppingCartRepository.save(shoppingCart));
+    }
+
+    @Override
+    public ShoppingCartDto getShoppingCart(Long id) {
+        ShoppingCart shoppingCart = getShoppingCartFromUserId(id);
+        return shoppingCartMapper.toDto(shoppingCart);
+    }
+
+    @Override
+    public void deleteByCartItemId(Long id, Long userId) {
+        ShoppingCart scByUserId = getShoppingCartFromUserId(userId);
+        Set<CartItem> cartItems = scByUserId.getCartItems();
+        Optional<CartItem> first = cartItems.stream()
+                .filter(cartItem -> Objects.equals(cartItem.getId(), id))
+                .findFirst();
+        first.ifPresent(cartItem -> cartItemService.deleteById(cartItem.getId()));
+        first.ifPresent(cartItems::remove);
+        shoppingCartRepository.save(scByUserId);
+    }
+
+    @Override
+    public void deleteByBookId(Long id, Long userId) {
+        ShoppingCart shoppingCartFromUserId = getShoppingCartFromUserId(userId);
+        Set<CartItem> cartItems = shoppingCartFromUserId.getCartItems();
+        Optional<CartItem> first = cartItems.stream()
+                .filter(cartItem -> Objects.equals(cartItem.getBook().getId(), id))
+                .findFirst();
+        first.ifPresent(cartItem -> cartItemService.deleteById(cartItem.getId()));
+    }
+
+    @Override
+    public void delete(Long id) {
+        ShoppingCart shoppingCartFromUserId = getShoppingCartFromUserId(id);
+        Set<CartItem> cartItems = shoppingCartFromUserId.getCartItems();
+        for (CartItem cartItem : cartItems) {
+            cartItemService.deleteById(cartItem.getId());
+        }
+        shoppingCartRepository.delete(shoppingCartFromUserId);
+    }
+
+    private User getUserById(Long id) {
+        return userRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("User with id: " + id
+                        + " not found in db")
+        );
+    }
+
+    @Override
+    public ShoppingCartDto updateCartItem(Long id, CartItemDtoUpdate cartItemDto) {

```

Comment:
Why method update cart item located in shopping cart service and returns shopping cart dto?